rust/worker/src/blockstore/arrow_blockfile/block/types.rs

```diff
@@ -222,3 +222,324 @@ impl BlockData {
         total_size
     }
 }
+
+// ============== BlockDataBuilder ==============
+
+enum KeyBuilder {
+    StringBuilder(StringBuilder),
+    FloatBuilder(Float32Builder),
+    BoolBuilder(BooleanBuilder),
+}
+
+enum ValueBuilder {
```

Comment:
Could we add a todo for other value types

Reply:
I have it coming in a stacked PR, was just waiting for these to get in before pushing more up